newlib/libc/machine/riscv/memcpy.c

```diff
@@ -58,33 +114,33 @@ memcpy(void *__restrict aa, const void *__restrict bb, size_t n)
   if (unlikely (lend - la > 8))
     {
       while (lend - la > 8)
-	{
-	  long b0 = *lb++;
-	  long b1 = *lb++;
-	  long b2 = *lb++;
-	  long b3 = *lb++;
-	  long b4 = *lb++;
-	  long b5 = *lb++;
-	  long b6 = *lb++;
-	  long b7 = *lb++;
-	  long b8 = *lb++;
-	  *la++ = b0;
-	  *la++ = b1;
-	  *la++ = b2;
-	  *la++ = b3;
-	  *la++ = b4;
-	  *la++ = b5;
-	  *la++ = b6;
-	  *la++ = b7;
-	  *la++ = b8;
-	}
+        {
+          long b0 = *lb++;
+          long b1 = *lb++;
+          long b2 = *lb++;
+          long b3 = *lb++;
+          long b4 = *lb++;
+          long b5 = *lb++;
+          long b6 = *lb++;
+          long b7 = *lb++;
+          long b8 = *lb++;
+          *la++ = b0;
+          *la++ = b1;
+          *la++ = b2;
+          *la++ = b3;
+          *la++ = b4;
+          *la++ = b5;
+          *la++ = b6;
+          *la++ = b7;
+          *la++ = b8;
+        }
     }
 
   while (la < lend)
     BODY (la, lb, long);
 
-  a = (char *)la;
-  b = (const char *)lb;
+  a = (unsigned char *)la;
+  b = (const unsigned char *)lb;
   if (unlikely (a < end))
     goto small;
```

Comment:
I am always in favor of getting rid of gotos if possible.
I suspect replacing this with a function call will not change the  output, but it makes debugging etc a lot nicer.